packages/messaging/src/Message.ts

```diff
@@ -26,312 +25,287 @@ import { DidResolver } from '@kiltprotocol/did'
-    receiverKeyUri: DidResourceUri,
-    {
-      resolver = DidResolver,
-    }: {
-      resolver?: IDidResolver
-    } = {}
-  ): Promise<IEncryptedMessage> {
-    const receiverKey = await resolver.resolveKey(receiverKeyUri)
-    if (!receiverKey) {
-      throw new SDKErrors.DidError(`Cannot resolve key "${receiverKeyUri}"`)
-    }
-    if (this.receiver !== receiverKey.controller) {
-      throw new SDKErrors.IdentityMismatchError(
-        'receiver public key',
-        'receiver'
-      )
-    }
-    if (this.sender !== senderDetails.uri) {
-      throw new SDKErrors.IdentityMismatchError('sender public key', 'sender')
-    }
-    const senderKey = Did.getKey(senderDetails, senderKeyId)
-    if (!senderKey || !encryptionKeyTypes.includes(senderKey.type)) {
-      throw new SDKErrors.DidError(
-        `Cannot find key with ID "${senderKeyId}" for the sender DID`
-      )
-    }
-    const senderKeyAlgType =
-      Did.Utils.getEncryptionAlgorithmForEncryptionKeyType(
-        senderKey.type as EncryptionKeyType
-      )
-    if (senderKeyAlgType !== 'x25519-xsalsa20-poly1305') {
-      throw new SDKErrors.EncryptionError(
-        'Only the "x25519-xsalsa20-poly1305" encryption algorithm currently supported'
-      )
+    if (sender !== senderKeyDetails.controller) {
+      throw new SDKErrors.IdentityMismatchError('Encryption key', 'Sender')
     }
 
-    const toEncrypt: IEncryptedMessageContents = {
-      body: this.body,
-      createdAt: this.createdAt,
-      sender: this.sender,
-      receiver: this.receiver,
-      messageId: this.messageId,
-      inReplyTo: this.inReplyTo,
-      references: this.references,
-    }
+    // checks the message body
+    errorCheckMessageBody(decrypted.body)
 
-    const serialized = stringToU8a(JSON.stringify(toEncrypt))
+    // checks the message structure
+    errorCheckMessage(decrypted)
+    // make sure the sender is the owner of the identity
+    ensureOwnerIsSender(decrypted)
 
-    const encrypted = await encryptCallback({
-      alg: senderKeyAlgType,
-      data: serialized,
-      publicKey: senderKey.publicKey,
-      peerPublicKey: receiverKey.publicKey,
+    return decrypted
+  } catch (cause) {
+    throw new SDKErrors.ParsingMessageError(undefined, {
+      cause: cause as Error,
     })
-    const ciphertext = u8aToHex(encrypted.data)
-    const nonce = u8aToHex(encrypted.nonce)
+  }
+}
 
-    return {
-      receivedAt: this.receivedAt,
-      ciphertext,
-      nonce,
-      senderKeyUri: `${senderDetails.uri}${senderKey.id}`,
-      receiverKeyUri: receiverKey.id,
-    }
+/**
+ * Constructs a message from a message body.
+ * This should be encrypted with [[Message.encrypt]] before sending to the receiver.
+ *
+ * @param body The body of the message.
+ * @param sender The DID of the sender.
+ * @param receiver The DID of the receiver.
+ * @returns The message created.
+ */
+export function fromBody(
+  body: MessageBody | CompressedMessageBody,
```

Comment:
The compression for messages seems to be not thought through. You can only compress the message body (not a full message) and if you try to make a new `Message` object from a compressed message body, it is decompressed, before making the message.
Seems like a compressed message body can never really be used for message sending.

Reply:
I would like to at least get rid of the option to give a compressed message body to this function. What d you think?

Reply:
If it is so half-baked, let’s rather drop it :+1:

Reply:
I also added a ticket for getting rid of compression altogether